src/main/java/org/jenkinsci/plugins/fodupload/steps/FortifyDastPipeline.java

```diff
@@ -868,23 +869,26 @@ private void saveWorkflowSiteScanSettings(FilePath workspace, PrintStream printS
     private void saveApiScanSettings(FilePath workspace, PrintStream printStream, DastScanSharedBuildStep dastScanSharedBuildStep) throws Exception {
 
         //TodO:-Change to switch
-
+        String sourceUrn = null;
         if (FodEnums.DastApiType.OpenApi.toString().equalsIgnoreCase(selectedApiType)) {
+            if(FodEnums.ApiSourceType.valueOf(openApiRadioSource) == FodEnums.ApiSourceType.FileId) {
+                PatchDastFileUploadResponse response = dastScanSharedBuildStep.DastManifestFileUpload(workspace, this.openApiFilePath,
+                        printStream, FodEnums.DastScanFileTypes.OpenAPIDefinition , dastScanSharedBuildStep.getFodApiConnection());
 
-            PatchDastFileUploadResponse response = dastScanSharedBuildStep.DastManifestFileUpload(workspace, this.openApiFilePath,
-                    printStream, FodEnums.DastScanFileTypes.OpenAPIDefinition , dastScanSharedBuildStep.getFodApiConnection());
+                if (response == null || !response.isSuccess || response.fileId <= 0) {
 
-            if (response == null || !response.isSuccess || response.fileId <= 0) {
+                    Utils.logger(printStream,
+                            String.format("Failed to upload payload for release Id %s", releaseId));
 
-                Utils.logger(printStream,
-                        String.format("Failed to upload payload for release Id %s", releaseId));
+                    throw new Exception(String.format("Failed to upload payload for release Id %s", releaseId));

```

Comment:
Check if the function that invokes this method has the catch block and the logger inside the catch, If so , please remove the Utils.logger from here to avoid dual entry

Reply:
No, It does not has try catch. Thanks!

Reply:
Please check the calling function at line number 1102 which call the SaveScanSetting function , which in turns call the saveApiScanSettings function , and there is catch block , 

Thanks,